Documentation/eliminating-pre-builts.md

```diff
@@ -27,42 +50,159 @@ To protect against this and catch any new pre-builts as soon as possible, Arcade
+
+```text
+1 new packages used not in baseline! See report at ./artifacts/source-build/self/prebuilt-report/baseline-comparison.xml for more information. Package IDs are:
+  System.Text.Json.8.0.0
+```
+
+You look at `./eng/SourceBuildPrebuiltBaseline.xml` and determine that `System.Text.Json.8.0.0` is not listed as an exclusion. You then examine `./prebuilt-usage.xml` and see the following:
+
+```xml
+<UsageData>
+  <Usages>
+    <Usage Id="System.Text.Json" Version="8.0.0" File=".../SomeProject/project.assets.json" IsDirectDependency="true">
+  </Usages>
+</UsageData>
+```
 
-### Pre-built through a transitive dependency
+The `project.assets.json` file shows the following:
 
-During a project dependency updated, a new pre-built might be introduced by a new or updated transitive dependency. While the Arcade tooling will highlight the name and version of the pre-built in the build exception as well as the project that restored the dependency in question, it will not point out where exactly in the dependency tree it is. This means that it's up to the developer to identify what type of dependency they are dealing with (direct or transitive) and choose the correct way of handling the issue. In case of a transitive dependency, it might be hard to identify the relationship that is bringing in the pre-built into the project, especially if the developer has limited knowledge of the project or code-base in general.
+```json
+{ 
+  "targets": {
+    "net8.0": {
+      "System.Text.Json/8.0.0": {
+        "type": "package",
+        "dependencies": {
+        }
+      }
+    }
+  }
+}
+```
 
-Arcade source-build infrastructure helps accomplish this this by pointing out the `project.assets.json` file that is referencing the pre-built in the `./artifacts/source-build/self/prebuild-report/prebuild-usage.xml` report file. A `project.assets.json` file is a NuGet restore process artifact that contains a resolved dependency tree for a specific project. Every package that was restored by a given project is mentioned there with links between the dependencies, allowing the reader to identify transitive dependencies and the direct dependencies referencing them.
+You find a package reference to `System.Text.Json.8.0.0`, like the one above, in one of your .csproj files:
 
-Example of a pre-built caused by a transitive dependency and corresponding entries in files mentioned above:
+```xml
+<ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="8.0.0" />
+</ItemGroup>
+```
 
-Exception identifying the pre-built:
+In this example, `System.Text.Json` is a direct dependency causing a pre-built.
+
+At this point, determine how to handle the dependency by following the [adding dependencies documentation](https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-dependencies.md#adding-dependencies). Based on what you determine, you will likely have to do one or more of the following:
```

Comment:
I think that these steps would be better explained in depth in the adding dependencies documentation but including them as options in the example is helpful. It would be up to the reader to read the adding dependencies documentation to determine which step to take.

Reply:
Actually, based on Matt's comment above, I think that these steps would be better served in the dependencies documentation, as you originally suggested. My thinking is also that if the steps ever changed, we'd have to alter documentation in two places, rather than one. I'm going to remove the examples.